src/Nethermind/Nethermind.Core2.Abstractions/Containers/Eth1Data.cs

```diff
@@ -65,7 +63,7 @@ public override bool Equals(object? obj)
         public bool Equals(Eth1Data? other)
         {
             return !(other is null) &&
-                BlockHash == other.BlockHash &&
+                BlockHash.Equals(other.BlockHash) &&
```

Comment:
don't we want simple equality on Root and Bytes32?

Reply:
Hash got removed in the spec changes and changed to Root and Bytes32 depending on context, and when implementing I didn't originally override the equality operator, so got warnings and went around and changed a bunch to .Equals() ... since then I have put them back in (but == just calls .Equals() anyway).

If you want for consistency I can style it one way or the other.